internal/service/pushbasedlogexport/resource.go

```diff
@@ -51,7 +53,18 @@ func (r *pushBasedLogExportRS) Create(ctx context.Context, req resource.CreateRe
 	connV2 := r.Client.AtlasV2
 	projectID := tfPlan.ProjectID.ValueString()
 	if _, err := connV2.PushBasedLogExportApi.CreatePushBasedLogConfiguration(ctx, projectID, logExportConfigReq).Execute(); err != nil {
-		resp.Diagnostics.AddError("eError when creating push-based log export configuration", err.Error())
+		resp.Diagnostics.AddError("Error when creating push-based log export configuration", err.Error())
+
+		logConfig, _, _ := connV2.PushBasedLogExportApi.GetPushBasedLogConfiguration(ctx, projectID).Execute()
+		if logConfig != nil && slices.Contains(failureStates, *logConfig.State) {
+			log.Printf("[INFO] Unconfiguring push-based log export for project due to create failure: %s", projectID)
+			if _, err := connV2.PushBasedLogExportApi.DeletePushBasedLogConfiguration(ctx, projectID).Execute(); err != nil {
+				resp.Diagnostics.AddError("Error when unconfiguring push-based log export configuration", err.Error())
+				return
+			}
+		}
```

Comment:
could we extract to a function so it does not diverge with identical logic below?

Reply:
done